packages/framework-types/src/envelope.ts

```diff
@@ -26,18 +26,28 @@ export interface ScheduledCommandEnvelope extends Envelope {
 
 export type SuperKindType = 'domain' | 'notification' | 'booster'
 
-export interface EventEnvelope extends Envelope {
+export interface EventStoreEntryEnvelope extends Envelope {
   typeName: string
   version: number
-  kind: 'event' | 'snapshot'
   superKind: SuperKindType
   entityID: UUID
   entityTypeName: string
   value: EventInterface | EntityInterface
   createdAt: string
-  snapshottedEventCreatedAt?: string
 }
 
+export interface NonPersistedEventEnvelope extends EventStoreEntryEnvelope {
+  kind: 'event'
+}
+export interface EventEnvelope extends NonPersistedEventEnvelope {
+  persistedAt: string
+}
+
+export interface EntitySnapshotEnvelope extends EventStoreEntryEnvelope {
+  kind: 'snapshot'
+  snapshottedEventCreatedAt: string
+  snapshottedEventPersistedAt: string
+}
```

Comment:
There's a new type semantics for event store entries:
* `NonPersistedEventEnvelope`: An event that has not been stored yet
* `EventEnvelope`: An event that has been stored (it has a `persistedAt` timestamp set)
* `EntitySnapshotEnvelope`: A dedicated snapshot envelope that forces the presence of `snapshottedEventCreatedAt` and `snapshottedEventPersistedAt`. These fields reference the latest event processed for the current snapshot.